### PR TITLE
Do not timeout podman/docker pull if timeout value is '0'

### PR DESCRIPTION
**Pull request by fultonj**
If user sets "docker_pull_timeout: '0'" then do not use the
timeout command when running podman/docker pull. Also, use
"timeout -s KILL"; without KILL, podman on RHEL8 beta does
not timeout and deployment can hang.

Related: https://bugzilla.redhat.com/show_bug.cgi?id=1670625

### CODE REVIEW

**Comment by fultonj** [comment]
@guits

**Comment by guits** [comment]
@fultonj please, add a sign off

File: roles/ceph-container-common/tasks/fetch_image.yml [hunk 1]
```diff
@@ -176,8 +176,14 @@
     - nfs_group_name in group_names
     - ceph_nfs_container_inspect_before_pull.get('rc') == 0
 
+- name: set_fact timeout_command
+  set_fact:
+    timeout_command: "{{ 'timeout -s KILL ' ~ docker_pull_timeout if (docker_pull_timeout != '0') else '' }}"
```
Comment (leseb) [inline roles/ceph-container-common/tasks/fetch_image.yml]: What's the default timeout here?

Comment (guits) [reply roles/ceph-container-common/tasks/fetch_image.yml]: `roles/ceph-defaults/defaults/main.yml:558:docker_pull_timeout: "300s"`

**Pull request merged**